src/sources/exec.rs

```diff
@@ -0,0 +1,843 @@
+use crate::async_read::VecAsyncReadExt;
+use crate::config::{DataType, GlobalOptions};
+use crate::internal_events::{ExecCommandExecuted, ExecTimeout};
+use crate::{
+    config::{log_schema, SourceConfig, SourceDescription},
+    event::Event,
+    internal_events::{ExecEventReceived, ExecFailed},
+    shutdown::ShutdownSignal,
+    Pipeline,
+};
+use bytes::Bytes;
+use derive_is_enum_variant::is_enum_variant;
+use futures::{FutureExt, SinkExt, StreamExt};
+use serde::{Deserialize, Serialize};
+use std::io::{Error, ErrorKind};
+use std::path::PathBuf;
+use std::process::ExitStatus;
+use std::task::Poll;
+use std::time::{Duration, Instant};
+use tokio::io::{AsyncRead, AsyncReadExt, BufReader};
+use tokio::process::Command;
+use tokio::sync::mpsc::{channel, Sender};
+use tokio::time;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+// TODO: add back when serde-rs/serde#1358 is addressed (same as syslog)
+// #[serde(deny_unknown_fields)]
+#[serde(default)]
+pub struct ExecConfig {
+    #[serde(flatten)]
+    pub mode: Mode,
+    pub command: String,
+    pub arguments: Option<Vec<String>>,
+    pub current_dir: Option<PathBuf>,
+    pub include_stderr: Option<bool>,
```

Comment:
Could this simply be a `bool` with a default to `true`, possibly with a `serde(skip_serializing_if = …)` if you don't want it to appear in default configs?

Reply:
Yes, thank you it is much better as a `bool`. I have changed it.